google-cloud-pubsub/lib/google/cloud/pubsub/service.rb

```diff
@@ -150,13 +157,26 @@ def publish topic, messages
               data: data, attributes: attributes)
           end
 
+          publish_messages topic, messages
+        end
+
+        ##
+        # Unlike publish, publish_messages takes an array of message grpc
```

Comment:
Publisher breaks an existing design layer in the code because it needs to calculate the size of the messages over the wire (`msg.to_proto.size`). Because of this, we needed to add a separate publish method to our internal service layer. Its possible this implementation detail is changed in a future refactoring, since this object is not part of the public API exposed to the users.